server/repository/migrations/postgres/2022-02-23T14-00_create_requisition_line_trigger/up.sql

```diff
@@ -1,3 +1,29 @@
-CREATE TRIGGER requisition_line_trigger
-  AFTER INSERT OR UPDATE OR DELETE ON requisition_line
-  FOR EACH ROW EXECUTE PROCEDURE update_changelog();
+CREATE FUNCTION upsert_requisition_line_changelog()
+RETURNS trigger AS
+$$
+  BEGIN
+    INSERT INTO changelog (table_name, record_id, row_action, name_id, store_id)
+      SELECT 'requisition_line', NEW.id, 'UPSERT', name_id, store_id FROM requisition WHERE id = NEW.requisition_id;
+    -- Ignored for AFTER trigger
+    RETURN NULL;
+  END;
+$$ LANGUAGE 'plpgsql';
+
+CREATE FUNCTION delete_requisition_line_changelog()
+RETURNS trigger AS
+$$
+  BEGIN
+    INSERT INTO changelog (table_name, record_id, row_action, name_id, store_id)
+      SELECT 'requisition_line', OLD.id, 'DELETE', name_id, store_id FROM requisition WHERE id = OLD.requisition_id;
```

Comment:
Is it possible for the requisition to be deleted before the lines? I assume not?

Reply:
As long as we enforce FK constraints, I didn't think it was possible

Reply:
Agreed, just occurred to me that this would insert nothing if the associated record wasn't found.
I think we are fine to rely on the referential constraint here.